us-weather-history/wunderground_parser.py

```diff
@@ -68,14 +87,23 @@ def parse_station(station):
                                              average_min_temp, average_max_temp,
                                              record_min_temp, record_max_temp,
                                              record_min_temp_year, record_max_temp_year,
-                                             actual_precipitation, average_precipitation,
-                                             record_precipitation]))
+                                             actual_precipitation,]))
+                    if average_precipitation:
+                        out_file.write(',{}'.format(average_precipitation))
+                    if record_precipitation:
+                        out_file.write(',{}'.format(record_precipitation))
+
                     out_file.write('\n')
                     current_date += timedelta(days=1)
+                    
                 except:
                     # If the web page is formatted improperly, signal that the page may need
                     # to be downloaded again.
                     try_again = True
+                    
+                    '''#Used to produce a mock-up .csv for KSAF
```

Comment:
No, it was just an easy way to make a KSAF csv file without having to manually enter the dates with incomplete/faulty data. I'll take it out this next pull request.